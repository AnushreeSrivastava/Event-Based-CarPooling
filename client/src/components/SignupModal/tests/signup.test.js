import React from 'react';
import Signup from "../../SignupModal";
import { mount, shallow, render } from 'enzyme';
import Modal from 'react-bootstrap/Modal';

const clickFn = jest.fn();

describe('Singup Modal', () => {
  it('should render Signup modal', () => {
    const component = shallow(<Signup/>);
    expect(component).toMatchSnapshot();
  });

  it('should have a boolean prop', () => {
    const wrapper = mount(<Signup show={false} onSubmit={clickFn}/>);
    expect(wrapper.props().show).toBe(false);
  });

  it('should render a signup modal', () => {
    const wrapper = shallow(<Signup show={false} onSubmit={clickFn}/>);
    const name = <Modal.Title>Sign-Up Form</Modal.Title>;
    expect(wrapper.contains(name)).toEqual(true);
  });  
});
