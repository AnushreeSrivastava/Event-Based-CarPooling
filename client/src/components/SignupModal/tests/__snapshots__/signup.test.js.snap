// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Singup Modal should render Signup 1`] = `ShallowWrapper {}`;

exports[`Singup Modal should render Signup modal 1`] = `ShallowWrapper {}`;
